src/components/Accordion/Accordion.tsx

```diff
@@ -8,12 +8,59 @@ import AccordionContent from './AccordionContent'
 import { DefaultBehavior } from '../../lib/accessibility'
 import { Accessibility } from '../../lib/accessibility/interfaces'
 
+export interface IAccordionPropsStrict {
+  /** An element type to render as (string or function). */
+  as?: any
+
+  /** Index of the currently active panel. */
+  activeIndex?: number[] | number
+
+  /** Primary content. */
+  children?: React.ReactNode
+
+  /** Additional classes. */
+  className?: string
+
+  /** Initial activeIndex value. */
+  defaultActiveIndex?: number[] | number
+
+  /** Only allow one panel open at a time. */
+  exclusive?: boolean
+
+  /**
+   * Called when a panel title is clicked.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All item props.
+   */
+  onTitleClick?: Function
+
+  /** Shorthand array of props for Accordion. */
+  panels?: {
+    content: React.ReactNode | object
+    title: React.ReactNode | object
+  }[]
+
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: object | Function
+
+  /** Custom styles to be applied for component. */
+  styles?: object | Function
+
+  /** Custom variables to be applied for component. */
+  variables?: object | Function
+}
+
+export interface IAccordionProps extends IAccordionPropsStrict {
```

Comment:
actually it can be done in more generic way

```TS
type Indexer<T> = { [key: string]: T }

interface IAccordionPropsStrict {
  as?: string
  ...
}

interface IAccordionProps extends Indexer<any>, IAccordionPropsStrict {
}
```

Reply:
I like these approach as being more generic, but I think the NotStrictProps types defined as is is sufficient enough and is more explicit with it's name when defining the not strict props interfaces. I will change the definition if you think that we should go with the Indexer type.